esp/esp/program/modules/handlers/volunteersignup.py

```diff
@@ -51,8 +53,17 @@ def module_properties(cls):
             "seq": 0,
             }
 
+    def require_auth(self):
+        if Tag.getBooleanTag('volunteer_require_auth', self.program, default=False):
+            return True
+        else:
+            return False
+
     @main_call
     @no_auth
+    @meets_deadline("/Signup", False)
+    #Would prefer something like @meets_deadline("/Signup", self.require_auth()),
```

Comment:
Ah, I see -- yeah, that won't work as such.  Probably you want to, in `check_deadline`, rather than passing in an arg, have it check `moduleObj.require_auth()`, on line `554`.  It means you have to say the method name is fixed, but that's probably a good idea anyway.  (Then you'll want to define it on `ProgramModuleObj` to match current default behavior.